<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1434</width>
    <height>875</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BMF Font Generator</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>:/images/logo.png</normaloff>:/images/logo.png</iconset>
  </property>
  <property name="documentMode">
   <bool>true</bool>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::TabShape::Rounded</enum>
  </property>
  <widget class="QWidget" name="central_widget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QStackedWidget" name="stacked_widget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="main_page">
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QTabWidget" name="fonts_tab_widget">
          <property name="currentIndex">
           <number>-1</number>
          </property>
          <property name="tabsClosable">
           <bool>true</bool>
          </property>
          <property name="movable">
           <bool>true</bool>
          </property>
          <property name="tabBarAutoHide">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="no_font_loaded_page">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="lbl_no_font_loaded">
          <property name="text">
           <string>No font loaded. Please create a new font or open an existing one.
                                            </string>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menu_bar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1434</width>
     <height>33</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_open_font"/>
    <addaction name="separator"/>
    <addaction name="action_save_font"/>
    <addaction name="separator"/>
    <addaction name="action_quit"/>
   </widget>
   <widget class="QMenu" name="menu_edit">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_help">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="action_about"/>
   </widget>
   <addaction name="menu_file"/>
   <addaction name="menu_edit"/>
   <addaction name="menu_help"/>
  </widget>
  <widget class="QToolBar" name="tool_bar">
   <property name="windowTitle">
    <string>Tools</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonStyle::ToolButtonTextUnderIcon</enum>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="action_open_font">
   <property name="icon">
    <iconset theme="document-open"/>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open a font.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_save_font">
   <property name="icon">
    <iconset theme="document-save"/>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save the current font.</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_about">
   <property name="icon">
    <iconset resource="../resources/GeneralResources.qrc">
     <normaloff>:/AppIcon512.png</normaloff>:/AppIcon512.png</iconset>
   </property>
   <property name="text">
    <string>About ...</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="action_export_font">
   <property name="icon">
    <iconset theme="export"/>
   </property>
   <property name="text">
    <string>Export</string>
   </property>
   <property name="toolTip">
    <string>Export the current font.</string>
   </property>
  </action>
  <action name="action_quit">
   <property name="icon">
    <iconset theme="application-exit"/>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_new_font">
   <property name="icon">
    <iconset theme="document-new">
     <normaloff>:/ui/icons-dark/document.svg</normaloff>:/ui/icons-dark/document.svg</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="toolTip">
    <string>Create a new font.</string>
   </property>
  </action>
  <action name="action_send_feedback">
   <property name="icon">
    <iconset theme="feedback">
     <normaloff>:/ui/icons-dark/message.svg</normaloff>:/ui/icons-dark/message.svg</iconset>
   </property>
   <property name="text">
    <string>Send Feedback</string>
   </property>
   <property name="toolTip">
    <string>Send feedback to the developer.</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../resources/GeneralResources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>action_about</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>show_about_app()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>549</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_export_font</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_export_font_action()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>595</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_open_font</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_open_font_action()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>595</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_save_font</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_save_font_action()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>595</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_new_font</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_new_font_action()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>595</x>
     <y>383</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fonts_tab_widget</sender>
   <signal>tabCloseRequested(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_tab_close_requested(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>716</x>
     <y>482</y>
    </hint>
    <hint type="destinationlabel">
     <x>716</x>
     <y>437</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_send_feedback</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>send_feedback()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>716</x>
     <y>437</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fonts_tab_widget</sender>
   <signal>currentChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_current_tab_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>716</x>
     <y>459</y>
    </hint>
    <hint type="destinationlabel">
     <x>716</x>
     <y>437</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_quit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>716</x>
     <y>437</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>show_about_app()</slot>
  <slot>send_feedback()</slot>
  <slot>on_open_settings_clicked()</slot>
  <slot>cancel_generation()</slot>
  <slot>on_new_font_action()</slot>
  <slot>on_open_font_action()</slot>
  <slot>on_save_font_action()</slot>
  <slot>on_export_font_action()</slot>
  <slot>on_tab_close_requested(int)</slot>
  <slot>on_current_tab_changed(int)</slot>
 </slots>
</ui>
